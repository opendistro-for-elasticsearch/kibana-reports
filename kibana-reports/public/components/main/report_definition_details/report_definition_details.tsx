/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React, { useEffect, useState } from 'react';
import {
  EuiFlexGroup,
  EuiFlexItem,
  EuiPage,
  EuiPageHeader,
  EuiTitle,
  EuiPageBody,
  EuiPageContent,
  EuiHorizontalRule,
  EuiSpacer,
  EuiPageHeaderSection,
  EuiButton,
  EuiText,
} from '@elastic/eui';
import { ReportDetailsComponent } from '../report_details/report_details';

export function ReportDefinitionDetails(props) {
  const [reportDefinitionDetails, setReportDefinitionDetails] = useState({});
  const reportDefinitionId = props.match['params']['reportDefinitionId'];

  const handleReportDefinitionDetails = (e) => {
    setReportDefinitionDetails(e);
  };

  const getReportDefinitionDetailsMetadata = (data) => {
    let reportDefinitionDetails = {
      name: data['report_name'],
      description: data['description'],
      created: data['time_created'],
      lastUpdated: data['last_updated'],
      source: data['report_source'],
      timePeriod:
        data['trigger']['trigger_params']['schedule']['interval']['period'] +
        ' ' +
        data['trigger']['trigger_params']['schedule']['interval']['unit'],
      fileFormat: data['report_params']['report_format'],
      reportHeader: '--',
      reportFooter: '--',
      triggerType: data['trigger']['trigger_type'],
      scheduleDetails: data['trigger']['trigger_params']['schedule_type'],
      alertDetails: '--',
      status: data['status'],
      deliveryChannels: data['delivery']['channel'],
      kibanaRecipients: data['delivery']['delivery_params']['recipients'],
      emailRecipients: '--', // todo: data model needs separate field for email vs kibana recipients
      emailSubject: data['delivery']['delivery_params']['subject'],
      emailBody: data['delivery']['delivery_params']['body'],
      includeReportAsAttachment:
        data['delivery']['delivery_params']['has_attachment'],
    };
    return reportDefinitionDetails;
  };

  useEffect(() => {
    const { httpClient } = props;
    httpClient
      .get(`../api/reporting/reportDefinitions/${reportDefinitionId}`)
      .then((response) => {
        handleReportDefinitionDetails(
          getReportDefinitionDetailsMetadata(response)
        );
      })
      .catch((error) => {
        console.error('error when getting report definition details:', error);
      });
  }, []);

  const includeReportAsAttachmentString = reportDefinitionDetails.includeReportAsAttachment
    ? 'True'
    : 'False';

  return (
    <EuiPage>
      <EuiPageBody>
        <EuiTitle size="l">
          <h1>Report definition</h1>
        </EuiTitle>
        <EuiSpacer size="m" />
        <EuiPageContent panelPaddingSize={'l'}>
          <EuiPageHeader>
            <EuiFlexItem>
              <EuiPageHeaderSection>
                <EuiTitle>
                  <h2>{reportDefinitionDetails.name}</h2>
                </EuiTitle>
              </EuiPageHeaderSection>
            </EuiFlexItem>
            <EuiFlexGroup
              justifyContent="flexEnd"
              alignItems="flexEnd"
              gutterSize="xs"
            >
              <EuiFlexItem />
              <EuiFlexItem />
              <EuiFlexItem />
              <EuiFlexItem />
              <EuiFlexItem />
              <EuiFlexItem>
                <EuiText size="xs">
                  <h2>
                    <a href="#">Disable</a>
                  </h2>
                  <div></div>
                </EuiText>
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiText size="xs">
                  <h2>
                    <a href="#">Enable</a>
                  </h2>
                  <div></div>
                </EuiText>
              </EuiFlexItem>
              <EuiFlexItem grow={false}>
                <EuiButton
                  fill={true}
                  onClick={() => {
                    window.location.assign('opendistro_kibana_reports#/edit');
                  }}
                >
                  Edit
                </EuiButton>
              </EuiFlexItem>
              <EuiFlexItem />
              <EuiFlexItem />
            </EuiFlexGroup>
          </EuiPageHeader>
          <EuiHorizontalRule />
          <EuiTitle>
            <h3>Report settings</h3>
          </EuiTitle>
          <EuiSpacer />
          <EuiFlexGroup>
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Name'}
              reportDetailsComponentContent={reportDefinitionDetails.name}
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Description'}
              reportDetailsComponentContent={
                reportDefinitionDetails.description
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Created'}
              reportDetailsComponentContent={reportDefinitionDetails.created}
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Last updated'}
              reportDetailsComponentContent={
                reportDefinitionDetails.lastUpdated
              }
            />
          </EuiFlexGroup>
          <EuiSpacer />
          <EuiFlexGroup>
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Source'}
              reportDetailsComponentContent={reportDefinitionDetails.source}
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Time period'}
              reportDetailsComponentContent={
                reportDefinitionDetails.timePeriod
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'File format'}
              reportDetailsComponentContent={
                reportDefinitionDetails.fileFormat
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Report header'}
              ReportDefinitionDetails={reportDefinitionDetails.reportHeader}
            />
          </EuiFlexGroup>
          <EuiSpacer />
          <EuiFlexGroup>
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Report footer'}
              reportDetailsComponentContent={
                reportDefinitionDetails.reportFooter
              }
            />
            <ReportDetailsComponent />
            <ReportDetailsComponent />
            <ReportDetailsComponent />
          </EuiFlexGroup>
          <EuiSpacer />
          <EuiTitle>
            <h3>Report trigger</h3>
          </EuiTitle>
          <EuiSpacer />
          <EuiFlexGroup>
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Trigger type'}
              reportDetailsComponentContent={
                reportDefinitionDetails.triggerType
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Schedule details'}
              reportDetailsComponentContent={
                reportDefinitionDetails.scheduleDetails
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Alert details'}
              reportDetailsComponentContent={
                reportDefinitionDetails.alertDetails
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Status'}
              reportDetailsComponentContent={reportDefinitionDetails.status}
            />
          </EuiFlexGroup>
          <EuiSpacer />
          <EuiTitle>
            <h3>Delivery settings</h3>
          </EuiTitle>
          <EuiSpacer />
          <EuiFlexGroup>
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Delivery channels'}
              reportDetailsComponentContent={
                reportDefinitionDetails.deliveryChannels
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Kibana recipients'}
              reportDetailsComponentContent={
                reportDefinitionDetails.kibanaRecipients
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Email recipients'}
              reportDetailsComponentContent={
                reportDefinitionDetails.emailRecipients
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Email subject'}
              reportDetailsComponentContent={
                reportDefinitionDetails.emailSubject
              }
            />
          </EuiFlexGroup>
          <EuiSpacer />
          <EuiFlexGroup>
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Email body'}
              reportDetailsComponentContent={
                reportDefinitionDetails.emailBody
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Include report as attachment'}
              reportDetailsComponentContent={includeReportAsAttachmentString}
            />
            <ReportDetailsComponent />
            <ReportDetailsComponent />
          </EuiFlexGroup>
        </EuiPageContent>
      </EuiPageBody>
    </EuiPage>
  );
}
