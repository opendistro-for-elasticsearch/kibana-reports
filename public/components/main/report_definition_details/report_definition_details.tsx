/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React, { useState } from 'react';
import {
  EuiFlexGroup,
  EuiFlexItem,
  EuiPage,
  EuiPageHeader,
  EuiTitle,
  EuiPageBody,
  EuiPageContent,
  EuiHorizontalRule,
  EuiSpacer,
  EuiPageHeaderSection,
  EuiButton,
  EuiText,
} from '@elastic/eui';
import { ReportDetailsComponent } from '../report_details/report_details';

const created_date = new Date('April 20, 2020 20:32:12');

const reportDefinitionDetailsMockMetadata = {
  name: '[Logs] Web traffic',
  description: '--',
  created: created_date.toString(),
  last_updated: '--',
  source: 'dashboards/daily_sales',
  time_period: 'Last 30 minutes',
  file_format: 'PDF',
  report_header: '--',
  report_footer: '--',
  trigger_type: 'Schedule',
  schedule_details: '--',
  alert_details: '--',
  status: 'Active',
  delivery_channels: ['Kibana reports'],
  kibana_recipients: ['admin'],
  email_recipients: 'user1@email.com',
  email_subject: 'Latest web traffic report',
  email_body:
    'View report details %REPORT_DETAILS_URL% Download report file %FILE_DOWNLOAD_URL%',
  include_report_as_attachment: true,
};

interface ReportDefinitionDetailsRouteProps {
  reportDefinitionId: string;
  reportDefinitionDetailsMetadata: {
    name: string;
    description: string;
    created: string;
    last_updated: string;
    source: string;
    time_period: string;
    file_format: string;
    report_header: string;
    report_footer: string;
    trigger_type: string;
    schedule_details: string;
    alert_details: string;
    status: string;
    delivery_channels: string[];
    kibana_recipients: string[];
    email_recipients: string;
    email_subject: string;
    email_body: string;
    include_report_as_attachment: boolean;
  };
}

export function ReportDefinitionDetails(
  props: ReportDefinitionDetailsRouteProps
) {
  const reportId = props.reportDefinitionId;

  const includeReportAsAttachmentString = reportDefinitionDetailsMockMetadata.include_report_as_attachment
    ? 'True'
    : 'False';

  return (
    <EuiPage>
      <EuiPageBody>
        <EuiTitle size="l">
          <h1>Report definition</h1>
        </EuiTitle>
        <EuiSpacer size="m" />
        <EuiPageContent panelPaddingSize={'l'}>
          <EuiPageHeader>
            <EuiFlexItem>
              <EuiPageHeaderSection>
                <EuiTitle>
                  <h2>{reportDefinitionDetailsMockMetadata.name}</h2>
                </EuiTitle>
              </EuiPageHeaderSection>
            </EuiFlexItem>
            <EuiFlexGroup
              justifyContent="flexEnd"
              alignItems="flexEnd"
              gutterSize="xs"
            >
              <EuiFlexItem></EuiFlexItem>
              <EuiFlexItem></EuiFlexItem>
              <EuiFlexItem></EuiFlexItem>
              <EuiFlexItem></EuiFlexItem>
              <EuiFlexItem></EuiFlexItem>
              <EuiFlexItem>
                <EuiText size="xs">
                  <h2>
                    <a href="#">Disable</a>
                  </h2>
                  <div></div>
                </EuiText>
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiText size="xs">
                  <h2>
                    <a href="#">Enable</a>
                  </h2>
                  <div></div>
                </EuiText>
              </EuiFlexItem>
              <EuiFlexItem grow={false}>
                <EuiButton 
                  fill={true}
                  onClick={() => {
                    window.location.assign('opendistro_kibana_reports#/edit');
                  }}
                >
                  Edit
                </EuiButton>
              </EuiFlexItem>
              <EuiFlexItem></EuiFlexItem>
              <EuiFlexItem></EuiFlexItem>
            </EuiFlexGroup>
          </EuiPageHeader>
          <EuiHorizontalRule />
          <EuiTitle>
            <h3>Report settings</h3>
          </EuiTitle>
          <EuiSpacer />
          <EuiFlexGroup>
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Name'}
              reportDetailsComponentContent={
                reportDefinitionDetailsMockMetadata.name
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Description'}
              reportDetailsComponentContent={
                reportDefinitionDetailsMockMetadata.description
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Created'}
              reportDetailsComponentContent={
                reportDefinitionDetailsMockMetadata.created
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Last updated'}
              reportDetailsComponentContent={
                reportDefinitionDetailsMockMetadata.last_updated
              }
            />
          </EuiFlexGroup>
          <EuiSpacer />
          <EuiFlexGroup>
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Source'}
              reportDetailsComponentContent={
                reportDefinitionDetailsMockMetadata.source
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Time period'}
              reportDetailsComponentContent={
                reportDefinitionDetailsMockMetadata.time_period
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'File format'}
              reportDetailsComponentContent={
                reportDefinitionDetailsMockMetadata.file_format
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Report header'}
              ReportDefinitionDetails={
                reportDefinitionDetailsMockMetadata.report_header
              }
            />
          </EuiFlexGroup>
          <EuiSpacer />
          <EuiFlexGroup>
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Report footer'}
              reportDetailsComponentContent={
                reportDefinitionDetailsMockMetadata.report_footer
              }
            />
            <ReportDetailsComponent />
            <ReportDetailsComponent />
            <ReportDetailsComponent />
          </EuiFlexGroup>
          <EuiSpacer />
          <EuiTitle>
            <h3>Report trigger</h3>
          </EuiTitle>
          <EuiSpacer />
          <EuiFlexGroup>
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Trigger type'}
              reportDetailsComponentContent={
                reportDefinitionDetailsMockMetadata.trigger_type
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Schedule details'}
              reportDetailsComponentContent={
                reportDefinitionDetailsMockMetadata.schedule_details
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Alert details'}
              reportDetailsComponentContent={
                reportDefinitionDetailsMockMetadata.alert_details
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Status'}
              reportDetailsComponentContent={
                reportDefinitionDetailsMockMetadata.status
              }
            />
          </EuiFlexGroup>
          <EuiSpacer />
          <EuiTitle>
            <h3>Delivery settings</h3>
          </EuiTitle>
          <EuiSpacer />
          <EuiFlexGroup>
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Delivery channels'}
              reportDetailsComponentContent={
                reportDefinitionDetailsMockMetadata.delivery_channels
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Kibana recipients'}
              reportDetailsComponentContent={
                reportDefinitionDetailsMockMetadata.kibana_recipients
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Email recipients'}
              reportDetailsComponentContent={
                reportDefinitionDetailsMockMetadata.email_recipients
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Email subject'}
              reportDetailsComponentContent={
                reportDefinitionDetailsMockMetadata.email_subject
              }
            />
          </EuiFlexGroup>
          <EuiSpacer />
          <EuiFlexGroup>
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Email body'}
              reportDetailsComponentContent={
                reportDefinitionDetailsMockMetadata.email_body
              }
            />
            <ReportDetailsComponent
              reportDetailsComponentTitle={'Include report as attachment'}
              reportDetailsComponentContent={includeReportAsAttachmentString}
            />
            <ReportDetailsComponent />
            <ReportDetailsComponent />
          </EuiFlexGroup>
        </EuiPageContent>
      </EuiPageBody>
    </EuiPage>
  );
}
