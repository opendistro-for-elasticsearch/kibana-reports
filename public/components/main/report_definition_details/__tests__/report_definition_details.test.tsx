/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { render } from '@testing-library/react';
import { ReportDefinitionDetails } from '../report_definition_details';
import httpClientMock from '../../../../../test/httpMockClient';
import 'babel-polyfill';

describe('<ReportDefinitionDetails /> panel', () => {
  const propsMock = {
    match: {
      params: {
        reportDefinitionId: jest.fn(),
      },
    },
  };

  const match = {
    params: {
      reportDefinitionId: '1',
    },
  };

  test('render component', async (done) => {
    const { container } = render(
      <ReportDefinitionDetails
        httpClient={httpClientMock}
        props={propsMock}
        match={match}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
    done();
  });
});
