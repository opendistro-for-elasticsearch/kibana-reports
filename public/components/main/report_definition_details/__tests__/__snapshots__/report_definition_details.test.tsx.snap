// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportDefinitionDetails /> panel render component 1`] = `
<div
  class="euiPage"
>
  <main
    class="euiPageBody"
  >
    <h1
      class="euiTitle euiTitle--large"
    >
      Report definition
    </h1>
    <div
      class="euiSpacer euiSpacer--m"
    />
    <div
      class="euiPanel euiPanel--paddingLarge euiPageContent"
    >
      <div
        class="euiPageHeader euiPageHeader--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <div
            class="euiPageHeaderSection"
          >
            <h2
              class="euiTitle euiTitle--medium"
            />
          </div>
        </div>
        <div
          class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsFlexEnd euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          />
          <div
            class="euiFlexItem"
          />
          <div
            class="euiFlexItem"
          />
          <div
            class="euiFlexItem"
          />
          <div
            class="euiFlexItem"
          />
          <div
            class="euiFlexItem"
          >
            <div
              class="euiText euiText--extraSmall"
            >
              <h2>
                <a
                  href="#"
                >
                  Disable
                </a>
              </h2>
              <div />
            </div>
          </div>
          <div
            class="euiFlexItem"
          >
            <div
              class="euiText euiText--extraSmall"
            >
              <h2>
                <a
                  href="#"
                >
                  Enable
                </a>
              </h2>
              <div />
            </div>
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <button
              class="euiButton euiButton--primary euiButton--fill"
              type="button"
            >
              <span
                class="euiButton__content"
              >
                <span
                  class="euiButton__text"
                >
                  Edit
                </span>
              </span>
            </button>
          </div>
          <div
            class="euiFlexItem"
          />
          <div
            class="euiFlexItem"
          />
        </div>
      </div>
      <hr
        class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginLarge"
      />
      <h3
        class="euiTitle euiTitle--medium"
      >
        Report settings
      </h3>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Name
            </dt>
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Description
            </dt>
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Created
            </dt>
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Last updated
            </dt>
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Source
            </dt>
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Time period
            </dt>
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              File format
            </dt>
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Report header
            </dt>
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Report footer
            </dt>
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            />
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            />
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            />
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <h3
        class="euiTitle euiTitle--medium"
      >
        Report trigger
      </h3>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Trigger type
            </dt>
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Schedule details
            </dt>
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Alert details
            </dt>
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Status
            </dt>
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <h3
        class="euiTitle euiTitle--medium"
      >
        Delivery settings
      </h3>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Delivery channels
            </dt>
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Kibana recipients
            </dt>
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Email recipients
            </dt>
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Email subject
            </dt>
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Email body
            </dt>
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            >
              Include report as attachment
            </dt>
            <dd
              class="euiDescriptionList__description"
            >
              False
            </dd>
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            />
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
        <div
          class="euiFlexItem"
        >
          <dl
            class="euiDescriptionList euiDescriptionList--row"
          >
            <dt
              class="euiDescriptionList__title"
            />
            <dd
              class="euiDescriptionList__description"
            />
          </dl>
        </div>
      </div>
    </div>
  </main>
</div>
`;
